### PR TITLE
new locale: Occitan (closes #2061)

### LINKED ISSUE #2061: Occitan language support
Hello,
I'm going to provide a PR to support Occitan.

### PR DESCRIPTION
**Pull request by Mejans**
Hello!
This is my first PR here. I struggle a bit to end it...
For the "match" part and a JS function called ordinalNumber in Localize.
If someone could give me some hints.

Thanks!

closes #2061

### CODE REVIEW

**Comment by Mejans** [comment]
Hello @kossnocorp @rikkalo may I ask who I can contact to get some help in adding a new locale?
Thanks!

**Comment by rikkalo** [comment]
Hi, @Mejans!
Thanks for your PR 👍 
You can ask me

**Comment by Mejans** [comment]
Thanks @rikkalo ,
well I'm not a JavaScript expert, I don't know how to end this PR so it can be added =/

**Comment by kossnocorp** [comment]
@Mejans please run `yarn build-snapshots` and commit the results. I would also appreciate if you can format the code using Prettier.

**Comment by tan75** [comment]
Hi @Mejans 
thank you for this PR! 
The code is failing on the below error - could you please double-check it? 
image

**Comment by tan75** [comment]
@Mejans 
as well, please run `yarn locale-snapshots` to update snapshots.

**Review by tan75 — CHANGES_REQUESTED** [review]
Please address my comments. Let us know if you have any questions!

**Comment by Mejans** [comment]
> Please address my comments. Let us know if you have any questions!

Hello! Thanks, I've just made some changes, it was thought with PHP on mind, my bad.
I'm not really good at Javascript, I hope this passes the test!

**Comment by tan75** [comment]
Hi @Mejans 
apologies for the delay - the tests are failing because this locale is missing the below file:
`_lib/match/index.js`

The instructions on how to create this file are below:
https://date-fns.org/v2.25.0/docs/I18n-Contribution-Guide#match

Could you please have a look into this again?

**Comment by tan75** [comment]
TODO: delete the branch https://github.com/date-fns/date-fns/pull/2698

**Comment by fturmel** [comment]
Hope that's OK, trying to help push this to the finish line.

- rebased onto current master
- converted to TypeScript
- created the `match` file as best as I could by referencing the work in `localize`. I don't speak the language but do speak French, so it wasn't too hard to track progress

@Mejans Can you review the work and the snapshot.md file, and let us know how it looks?

**Comment by Mejans** [comment]
@fturmel d'accord :) Merci beaucoup, je vais regarder les fichiers pour vous dire si ça me semble bon. Encore merci

File: src/locale/oc/_lib/formatDistance/index.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    other: '{{count}} ans',
+  },
+
+  overXYears: {
+    one: 'mai d’un an',
+    other: 'mai de {{count}} ans',
+  },
+
+  almostXYears: {
+    one: 'gaireben an',
```
Comment (Mejans) [inline src/locale/oc/_lib/formatDistance/index.ts:84]: missing "un" between gaireben and an.

File: src/locale/oc/snapshot.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+| 1999-12-31T23:15:00.000Z | 45 minutas  | fa 45 minutas      | 1 ora                          |
+| 1999-12-31T23:00:00.000Z | 1 ora       | fa 1 ora           | 1 ora                          |
+| 1999-12-31T18:00:00.000Z | 6 oras      | fa 6 oras          | 6 oras                         |
+| 1999-12-30T00:00:00.000Z | 2 jorns     | fa 2 jorns         | 48 oras                        |
+| 1999-12-15T00:00:00.000Z | 17 jorns    | fa 17 jorns        | 408 oras                       |
+| 1999-12-01T00:00:00.000Z | 1 mes       | fa 1 mes           | 744 oras                       |
+| 1999-11-01T00:00:00.000Z | 2 meses     | fa 2 meses         | 1464 oras                      |
+| 1999-06-01T00:00:00.000Z | 7 meses     | fa 7 meses         | 5136 oras                      |
+| 1999-01-01T00:00:00.000Z | 1 an        | fa 1 an            | 8760 oras                      |
+| 1998-12-01T00:00:00.000Z | 1 an        | fa 1 an            | 9504 oras                      |
```
Comment (Mejans) [inline src/locale/oc/snapshot.md:284]: hum, maybe "1 e mièg" if possible or "1,5 an"

File: src/locale/oc/snapshot.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+If now is January 1st, 2000, 00:00.
+
+| Date                     | Result      | `addSuffix: true`  | With forced unit (i.e. `hour`) |
+| ------------------------ | ----------- | ------------------ | ------------------------------ |
+| 2006-01-01T00:00:00.000Z | 6 ans       | d’aquí 6 ans       | 52608 oras                     |
+| 2005-01-01T00:00:00.000Z | 5 ans       | d’aquí 5 ans       | 43848 oras                     |
+| 2004-01-01T00:00:00.000Z | 4 ans       | d’aquí 4 ans       | 35064 oras                     |
+| 2003-01-01T00:00:00.000Z | 3 ans       | d’aquí 3 ans       | 26304 oras                     |
+| 2002-01-01T00:00:00.000Z | 2 ans       | d’aquí 2 ans       | 17544 oras                     |
+| 2001-06-01T00:00:00.000Z | 1 an        | d’aquí 1 an        | 12408 oras                     |
```
Comment (Mejans) [inline src/locale/oc/snapshot.md:251]: if possible "1,5 ans" in place of "1 an"

File: src/locale/oc/_lib/formatDistance/index.ts (hunk 1 above)
Comment (fturmel) [reply src/locale/oc/_lib/formatDistance/index.ts]: Made the correction

File: src/locale/oc/snapshot.md [hunk 4]
```diff
@@ -0,0 +1,330 @@
+# Occitan (oc) locale
+
+## `format` and `parse`
+
+| Title                           | Token string | Date                     | `format` result                                   | `parse` result           |
+| ------------------------------- | ------------ | ------------------------ | ------------------------------------------------- | ------------------------ |
+| Calendar year                   | yo           | 1987-02-11T12:13:14.015Z | 1987ena                                           | 1987-01-01T00:00:00.000Z |
```
Comment (Mejans) [inline src/locale/oc/snapshot.md:7]: if the excepted output was the year, then it's not correct. Should only be "1987", it was excepted to be "the 1987th year", then it's correct

File: src/locale/oc/snapshot.md (hunk 3 above)
Comment (fturmel) [reply src/locale/oc/snapshot.md:251]: Don't think that's possible, the function truncates to the largest unit.

**Comment by fturmel** [comment]
@Mejans It might help if you compare to the more thoroughly tested outputs of the [en-US](https://github.com/date-fns/date-fns/blob/master/src/locale/en-US/snapshot.md) or [fr](https://github.com/date-fns/date-fns/blob/master/src/locale/fr/snapshot.md) locales

**Comment by Mejans** [comment]
sure ah ah ah, will avoid questioning ah ah.
Can I do it for tomorrow?

File: src/locale/oc/snapshot.md [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+If now is January 1st, 2000, 00:00.
+
+| Date                     | Result                            |
+| ------------------------ | --------------------------------- |
+| 2000-01-10T00:00:00.000Z | 10/01/2000                        |
+| 2000-01-05T00:00:00.000Z | dimècres a 00:00                  |
+| 2000-01-02T00:00:00.000Z | deman a 00:00                     |
+| 2000-01-01T00:00:00.000Z | uèi a 00:00                       |
+| 1999-12-31T00:00:00.000Z | ièr a 00:00                       |
+| 1999-12-27T00:00:00.000Z | lo diluns passat a 12946252800000 |
```
[the remark this answers is not in the document]
Comment (fturmel) [reply src/locale/oc/snapshot.md:303]: > lo diluns passat a 12946252800000

Definitely something off here. It's coming from `formatRelative` last week format: `'lo' eeee 'passat a' LT`

Let me know what the correct value should be and I'll update it.

[the remark this answers is not in the document]
Comment (fturmel) [reply src/locale/oc/snapshot.md:303]: In French, it's: `eeee 'dernier à' p,`, I suppose we just need to change `LT` to `p`. Let me just do that for now.

**Comment by fturmel** [comment]
> Can I do it for tomorrow?

Of course, no rush. This PR was dormant for over a year. I'm just stepping in to help a bit.

Ce serait dommage de perdre tout ce travail!

File: src/locale/oc/snapshot.md [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+|                                 | pppp         | 1987-02-11T12:13:14.015Z | 12:13:14 GMT+00:00                                | Errored                  |
+|                                 |              | 1453-05-29T23:59:59.999Z | 23:59:59 GMT+00:00                                | Errored                  |
+| Combination of date and time    | Pp           | 1987-02-11T12:13:14.015Z | 11/02/1987, 12:13                                 | 1987-02-11T12:13:00.000Z |
+|                                 |              | 1453-05-29T23:59:59.999Z | 29/05/1453, 23:59                                 | 1453-05-29T23:59:00.000Z |
+|                                 | PPpp         | 1987-02-11T12:13:14.015Z | 11 febr. 1987, 12:13:14                           | 1987-02-11T12:13:14.000Z |
+|                                 |              | 1453-05-29T23:59:59.999Z | 29 mai 1453, 23:59:59                             | 1453-05-29T23:59:59.000Z |
+|                                 | PPPppp       | 1987-02-11T12:13:14.015Z | 11 de febrièr 1987 a 12:13:14 GMT+0               | Errored                  |
```
[the remark this answers is not in the document]
Comment (fturmel) [reply src/locale/oc/snapshot.md:183]: We can ignore this, it's a larger issue either with snapshot generation or a limitation of the parser. I think most (if not all) locales error out here. Something to look at eventually.

File: src/locale/oc/snapshot.md (hunk 4 above)
Comment (fturmel) [reply src/locale/oc/snapshot.md:7]: Yes, `yo` is calendar year ordinal, so it does mean 1987th.

**Comment by Mejans** [comment]
Merci de l'avoir réveillé surtout ! Ça me faisait un peu de peine de savoir que j'avais une PR en attente, mais je ne savais pas par quoi commencer. 1 000 mercis

File: src/locale/oc/_lib/formatRelative/index.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import type { FormatRelativeFn } from '../../../types'
+
+const formatRelativeLocale = {
+  lastWeek: "'lo' eeee 'passat a' p",
```
Comment (Mejans) [inline src/locale/oc/_lib/formatRelative/index.ts:4]: remove the 'lo' here

File: src/locale/oc/_lib/localize/index.ts [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+import buildLocalizeFn from '../../../_lib/buildLocalizeFn/index'
+
+const eraValues = {
+  narrow: ['ab. J.C.', 'apr. J.C.'] as const,
+  abbreviated: ['ab. J.C.', 'apr. J.C.'] as const,
+  wide: ['abans Jèsus-Crist', 'après Jèsus-Crist'] as const,
+}
+
+const quarterValues = {
+  narrow: ['1', '2', '3', '4'] as const,
```
Comment (Mejans) [inline src/locale/oc/_lib/localize/index.ts:12]: almost like in French better:
  narrow: ['T1', 'T2', 'T3', 'T4'] as const,
  abbreviated: ['1èr trim.', '2nd trim.', '3en trim.', '4en trim.'] as const,

File: src/locale/oc/_lib/formatLong/index.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import type { FormatLong } from '../../../types'
+import buildFormatLongFn from '../../../_lib/buildFormatLongFn/index'
+
+const dateFormats = {
+  full: "EEEE, d 'de' MMMM y",
```
Comment (Mejans) [inline src/locale/oc/_lib/formatLong/index.ts:5]: remove the "," after EEEE, like in French

**Comment by Mejans** [comment]
Done!

**Comment by fturmel** [comment]
Thanks for reviewing all this, last changes were made. Seems like we're good to go?

**Review by fturmel — APPROVED** [review]

**Comment by Mejans** [comment]
Yes, brilliant, many thanks!

**Pull request merged**